### PR TITLE
[#76] Change toString in VariablesBinding to not showing basic types …

### LINKED ISSUE #76: Do not show basic type of variables when logging variable binding
Current log example:
```
27-05-2022 14:51:06.013 [http-nio-8081-exec-5] TRACE c.c.s.e.ExecutionEngineImpl - Using input merged context Context 766311399[ 
	 varBindings = {_pId=generateEccairsForms, enableQuestionLinks=false^^http://www.w3.org/2001/XMLSchema#boolean, questionTemplatesCacheUri=http://onto.fel.cvut.cz/ontologies/forms/eccairs/question-templates, enableEventTypeFiltering=true^^http://www.w3.org/2001/XMLSchema#boolean, eccairsMappingServiceUri=http://localhost:18080/rdf4j-server/repositories/eccairs-aviation-3.4.0.2?default-graph-uri=http%3A%2F%2Fonto.fel.cvut.cz%2Fontologies%2Feccairs%2Feccairs-3.4.0.2%2Fmapping, eccairsRepositoryUri=http://localhost:18080/rdf4j-server/repositories/eccairs-aviation-3.4.0.2, repositoryUrl=http://localhost:18080/rdf4j-server/repositories/form-generator, organizationType=, reportGraphId=http://www.inbas.cz/ontologies/reporting-tool/formGen1653655847997, dataGraphId=http://www.inbas.cz/ontologies/reporting-tool/formGen1653655848005, eccairsMappingUri=http://onto.fel.cvut.cz/ontologies/eccairs/eccairs-3.4.0.2/mapping, questionTemplatesCacheServiceUri=http://localhost:18080/rdf4j-server/repositories/form-generator?default-graph-uri=http%3A%2F%2Fonto.fel.cvut.cz%2Fontologies%2Fforms%2Feccairs%2Fquestion-templates, rdf4jServerUri=http://localhost:18080/rdf4j-server, eccairsVersion=3.4.0.2}
	 modelSize = 15774]
```

We would like to substitute e.g.:
`enableQuestionLinks=false^^http://www.w3.org/2001/XMLSchema#boolean` --> `enableQuestionLinks=false`

String literals should be used with quotes `"` and URIs with `<>.`

A/C:
- implement the change
- show here what it outputs for string literal, URIs, boolean value, int value

### PR DESCRIPTION
**Pull request by palagdan**
…of variables

Resolves #76 

Screenshot from 2024-07-31 09-53-31

In the picture, there are examples of string, int, boolean and Uri bindings

### CODE REVIEW

File: s-pipes-core/src/main/java/cz/cvut/spipes/engine/VariablesBinding.java [hunk 1]
```diff
@@ -179,7 +179,28 @@ public void load(final InputStream is, final String lang) throws IOException {
 
     @Override
     public String toString() {
-        return binding.asMap().toString();
+        StringBuilder sb = new StringBuilder();
```
Comment (blcham) [inline s-pipes-core/src/main/java/cz/cvut/spipes/engine/VariablesBinding.java:182]: I would use rather stream API and specific function:
`private static toString(RDFNode node)`

The implementation could look like:
return binding.asMap().entrySet().stream().map((k,v) -> new EntrySet(k, toString(v)).toString();

Maybe search if toString(RDFNode) is not a directly a function in RDFNode.

**Review by blcham — CHANGES_REQUESTED** [review]
See my comments

File: s-pipes-core/src/main/java/cz/cvut/spipes/engine/VariablesBinding.java [hunk 2]
```diff
@@ -179,9 +179,24 @@ public void load(final InputStream is, final String lang) throws IOException {
...
+        if (node.isLiteral()) {
+            return "\"" + node.asLiteral().getLexicalForm() + "\"";
+        } else if (node.isURIResource() || node.isResource()) {
+            return "<" + node.asResource().getURI() + ">";
+        } else {
+            return node.toString();
+        }
     }
 
+
```
Comment (blcham) [inline s-pipes-core/src/main/java/cz/cvut/spipes/engine/VariablesBinding.java:199]:
```suggestion
```

File: s-pipes-core/src/main/java/cz/cvut/spipes/engine/VariablesBinding.java [hunk 3]
```diff
@@ -179,9 +179,24 @@ public void load(final InputStream is, final String lang) throws IOException {
 
     @Override
     public String toString() {
-        return binding.asMap().toString();
+        return binding.asMap()
+                .entrySet()
+                .stream()
+                .map(entry -> entry.getKey() + "=" + toString(entry.getValue()))
+                .collect(Collectors.joining(", ", "{", "}"));
```
Comment (blcham) [inline s-pipes-core/src/main/java/cz/cvut/spipes/engine/VariablesBinding.java:186]: Not sure but isn't it same as:
```suggestion
        return binding.asMap()
                    .entrySet()
                    .stream()
                    .collect(Collectors.toMap(
                         Map.Entry::getKey, 
                         e -> toString(e.getValue())
                     ))
                     .toString();
```

**Review by blcham — APPROVED** [review]
@palagdan looks nice now, I have one more suggestion, which I did not however test ... if you like it more apply it otherwise leave it as it is and you can merge this PR, **but squash commits first**.

File: s-pipes-core/src/main/java/cz/cvut/spipes/engine/VariablesBinding.java (hunk 1 above)
Comment (blcham) [reply s-pipes-core/src/main/java/cz/cvut/spipes/engine/VariablesBinding.java]: done.

File: s-pipes-core/src/main/java/cz/cvut/spipes/engine/VariablesBinding.java (hunk 2 above)
Comment (palagdan) [reply s-pipes-core/src/main/java/cz/cvut/spipes/engine/VariablesBinding.java]: done

File: s-pipes-core/src/main/java/cz/cvut/spipes/engine/VariablesBinding.java (hunk 3 above)
Comment (palagdan) [reply s-pipes-core/src/main/java/cz/cvut/spipes/engine/VariablesBinding.java]: done

**Pull request merged**